staging/src/k8s.io/kubectl/pkg/kuberc/kuberc.go

```diff
@@ -0,0 +1,334 @@
+			continue
+		}
+		err = foundAliasCmd.Flags().Set(fl.Name, fl.Default)
+		if err != nil {
+			return args, fmt.Errorf("could not apply value %s to flag %s in alias %s err: %w", fl.Default, fl.Name, args[0], err)
+		}
+	}
+
+	// all args defined in kuberc should be appended to actual args.
+	args = append(args, aliasArgs.args...)
+	// Cobra (command.go#L1078) appends only root command's args into the actual args and ignores the others.
+	// We are appending the additional args defined in kuberc in here and
+	// expect that it will be passed along to the actual command.
+	rootCmd.SetArgs(args[1:])
+	return args, nil
+}
+
+// DefaultGetPreferences returns KubeRCConfiguration.
+// If users sets kuberc file explicitly in --kuberc flag, it has the highest
+// priority. If not specified, it looks for in KUBERC environment variable.
+// If KUBERC is also not set, it falls back to default .kuberc file at the same location
+// where kubeconfig's defaults are residing in.
+func DefaultGetPreferences(kuberc string) (*config.Preferences, error) {
+	kubeRCFile := RecommendedKubeRCFile
+	explicitly := false
+	if kuberc != "" {
+		kubeRCFile = kuberc
+		explicitly = true
+	}
+
+	if kubeRCFile == "" && os.Getenv("KUBERC") != "" {
+		kubeRCFile = os.Getenv("KUBERC")
+		explicitly = true
+	}
+
+	kubeRCBytes, err := os.ReadFile(kubeRCFile)
+	if err != nil {
+		if os.IsNotExist(err) && !explicitly {
+			return nil, nil
+		}
+		return nil, err
+	}
+	pref := &config.Preferences{}
+	_, gvk, err := scheme.Codecs.UniversalDecoder(v1alpha1.SchemeGroupVersion).Decode(kubeRCBytes, nil, pref)
+	if err != nil {
+		return nil, fmt.Errorf("could not be decoded gvk %s, err: %w", gvk, err)
+	}
+	return pref, nil
+}
+
+// Normally, we should extract this value directly from kuberc flag.
+// However, flag values are set during the command execution and
+// we are in very early stages to prepare commands prior to execute them.
+// Besides, we only need kuberc flag value in this stage.
+func getExplicitKuberc(args []string) string {
+	var kubercPath string
+	for i, arg := range args {
+		if arg == "--kuberc" {
+			if i+1 < len(args) {
+				kubercPath = args[i+1]
+				break
+			}
+		} else if strings.Contains(arg, "--kuberc=") {
+			parg := strings.Split(arg, "=")
+			if len(parg) > 1 {
+				kubercPath = parg[1]
+				break
+			}
+		}
+	}
+
+	if kubercPath == "" {
+		return ""
+	}
+
+	return kubercPath
+}
+
+// searchInArgs searches the given key in the args and returns
+// true, if it finds. Otherwise, it returns false.
+func searchInArgs(flagName string, shorthand string, args []string) bool {
+	for _, arg := range args {
+		// if flag is set in args in "--flag value" or "--flag=value" format,
+		// we should return it as found
+		if fmt.Sprintf("--%s", flagName) == arg || strings.HasPrefix(arg, fmt.Sprintf("--%s=", flagName)) {
+			return true
+		}
+		if shorthand != "" {
+			// shorthand can only be in "-n value" format
+			if fmt.Sprintf("-%s", flagName) == arg {
```

Comment:
This will cover almost every cases, apart from `-oyaml`. Because there is no mechanism that we can identify `-oyaml` is actually `--output yaml` or `-oyaml` shorthand flag for a totally different flag.

Reply:
Closing as mentioned in https://github.com/kubernetes/kubernetes/pull/125230#discussion_r1684878809